### PR TITLE
Increase test coverage for Cigar components

### PR DESCRIPTION
**Pull request by magicDGS**
### Description

Includes full coverage for `CigarElement` and `CigarOperator`, and new tests for `Cigar` (including explicit tests for already covered methods).

### Checklist

- [x] Code compiles correctly
- [x] New tests covering changes and new functionality
- [x] All tests passing
- [x] Extended the README / documentation, if necessary
- [ ] Is not backward compatible (breaks binary or source compatibility)

### CODE REVIEW

File: src/test/java/htsjdk/samtools/util/CigarElementUnitTest.java [hunk 1]
```diff
@@ -12,4 +14,28 @@
...
+                {mElement, new CigarElement(mElement.getLength(), CigarOperator.X), false},
+                // different class
+                {mElement, mElement.toString(), false}
+        };
+    }
+
+    @Test(dataProvider = "elementsForEquals")
+    public void testEqualsAndHashCode(final CigarElement element, final Object other, final boolean isEquals) {
+        Assert.assertEquals(element.equals(other), isEquals);
+        Assert.assertEquals(element.hashCode() == other.hashCode(), isEquals);
```
Comment (lbergelson) [inline src/test/java/htsjdk/samtools/util/CigarElementUnitTest.java]: I think this test isn't technically correct.  It's possible 
for `A != B` but `A.hashCode() == B.hashCode()`.   I think the correct assertion for the the hashCode part should be.

```
if( element.equals(other)) {
 Assert.assertEquals(element.hashCode(), other.hashCode());
}
```

Comment (magicDGS) [inline src/test/java/htsjdk/samtools/util/CigarElementUnitTest.java]: True! My bad! But in the condition I used the expected isEquals instead.

**Comment by magicDGS** [comment]
Addressed the comment, @lbergelson!

**Comment by magicDGS** [comment]
Thanks @lbergelson!

**Pull request merged**